gremlin/value.go

```diff
@@ -0,0 +1,156 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package gremlin
+
+import (
+	"fmt"
+	"strconv"
+)
+
+// ValueString a value used within query constructs
+type ValueString string
+
+// newValueStringFromArgument via inferance creates a correct ValueString
+func NewValueStringFromArgument(v interface{}) ValueString {
+	switch v := v.(type) {
+	case ValueString:
+		return v
+	case string:
+		return Quote(v)
+	case int:
+		return ValueString(strconv.Itoa(v))
+	default:
+		panic(fmt.Sprintf("argument %v: type %T not supported", v, v))
+	}
+}
+
+// String converts value to string
+func (v ValueString) String() string {
+	return string(v)
+}
+
+// DESC const definition
+const DESC = ValueString("DESC")
+
+// Quote used to quote string values as needed by query
+func (v ValueString) Quote() ValueString {
+	return ValueString(fmt.Sprintf(`"%s"`, v))
+}
+
+// Regex used for constructing a regexp expression string
+func (v ValueString) Regex() ValueString {
+	return ValueString(fmt.Sprintf("Regex(%s)", v.Quote()))
+}
+
+// StartsWith construct a regexp representing all that start with string
+func (v ValueString) StartsWith() ValueString {
+	return ValueString(fmt.Sprintf("%s.*", v)).Regex()
+}
+
+// EndsWith construct a regexp representing all that end with string
+func (v ValueString) EndsWith() ValueString {
+	return ValueString(fmt.Sprintf(".*%s", v)).Regex()
+}
+
+// Quote used to quote string values as needed by query
+func Quote(s string) ValueString {
+	return ValueString(s).Quote()
+}
+
+// Regex used for constructing a regexp expression string
+func Regex(s string) ValueString {
+	return ValueString(s).Regex()
+}
+
+// StartsWith construct a regexp representing all that start with string
+func StartsWith(s string) ValueString {
+	return ValueString(s).StartsWith()
+}
+
+// EndsWith construct a regexp representing all that end with string
+func EndsWith(s string) ValueString {
+	return ValueString(s).EndsWith()
+}
+
+func newValueString(name string, list ...interface{}) ValueString {
+	s := fmt.Sprintf("%s(", name)
+	first := true
+	for _, v := range list {
+		if !first {
+			s = s + ", "
+		}
+		first = false
+		s = s + NewValueStringFromArgument(v).String()
+	}
+	return ValueString(s + ")")
+}
+
+// Between append a Between() operation to query
+func Between(list ...interface{}) ValueString {
+	return newValueString("Between", list...)
+}
+
+// Contains append a Contains() operation to query
+func Contains(v interface{}) ValueString {
+	return newValueString("Contains", v)
```

Comment:
Contains predicate does not exist any more

Reply:
ok, implemented using Regex()